src/libraries/System.Private.CoreLib/src/System/Globalization/Ordinal.cs

```diff
@@ -277,6 +241,29 @@ internal static int IndexOfOrdinalIgnoreCase(ReadOnlySpan<char> source, ReadOnly
             return OrdinalCasing.IndexOf(source, value);
         }
 
+        internal static int LastIndexOfNoIgnoreCase(string source, string value, int startIndex, int count)
+        {
+            // startIndex is the index into source where we start search backwards from.
+            // leftStartIndex is the index into source of the start of the string that is
+            // count characters away from startIndex.
+            int leftStartIndex = startIndex - count + 1;
```

Comment:
This method could be rewritten in terms of span, which would also take advantage of span's SIMD optimizations.

```cs
int result = source.AsSpan(startIndex, count).LastIndexOf(value);
if (result >= 0) { result += startIndex; } // if match found, adjust 'result' by the actual start position
return result;
```